src/gistutils.jl

```diff
@@ -0,0 +1,104 @@
+"""
+    gist_from_file(filepath::AbstractString; private::Bool = true) -> fileurl::String
+
+Create a new gist from file at `filepath`. Return the "raw" HTTPS URL of the file.
+"""
+function gist_from_file(filepath::AbstractString; private::Bool = true)
+    @assert isfile(filepath)
+    # Using `with_new_gist` utility defined below instead of simply calling `gh gist create
+    # $filepath`. This seems to be required when the file at `filepath` is not a text file.
+    repo_http = with_new_gist(; private = private) do git_dir
+        cp(filepath, joinpath(git_dir, basename(filepath)))
+    end
+    return rstrip(repo_http, '/') * "/raw/" * basename(filepath)
+end
+
+"""
+    with_new_gist(f; private::Bool = true) -> repo_http::AbstractString
+
+Create a new gist, check it out as a local git repository at a temporary directory
+`git_dir`, call `f(git_dir)`, and then push it to remote.  Return the canonical gist HTTPS
+URL.
+"""
+function with_new_gist(f; private::Bool = true)
+    cmd = gh_cli_jll.gh()
+    cmd = `$cmd gist create`
+    if !private
+        cmd = `$cmd --public`
+    end
+    repo_http = chomp(String(communicate(cmd, "dummy")))
+
+    m = match(r"https://gist.github.com/(.*)", repo_http)
+    if m === nothing
+        error("Unrecognized output from `gh cli`: ", repo_http)
+    end
+    slug = m[1]
+    git_url = "git@gist.github.com:$slug"
+
+    response = Ref{HTTP.Response}()
+    @sync begin
+        # Get https://gist.github.com/$user/$slug from https://gist.github.com/$slug
+        @async response[] = HTTP.head(repo_http; redirect = false)
+
+        mktempdir() do git_dir
+            git(args) = run(`git -C $git_dir $args`)
+            git(`clone $git_url .`)
+            git_empty_history(git_dir)
+            f(git_dir)
+            git(`add .`)
+            git(`commit --allow-empty -m "Add files"`)
+            git(`push origin --force-with-lease`)
+        end
+    end
+
+    return get(Dict(response[].headers), "Location", repo_http)
+end
+
+"""
+    git_empty_history(git_dir)
+
+Empty the history of the current branch. Create an empty commit to start fresh.
+"""
+function git_empty_history(git_dir)
+    git(args) = run(`git -C $git_dir $args`)
```

Comment:
We have [`Git.jl`](https://github.com/JuliaVersionControl/Git.jl) instead of relying on git to be available in the system